platform/src/index.js

```diff
@@ -0,0 +1,23 @@
+
+/*global TOKEN_SERVER_URL -- is set by environment variable*/
+import { EducationPlatform } from "./EducationPlatform.js";
+
+const TOKEN_HANDLER_URL = TOKEN_SERVER_URL || "http://127.0.0.1:10000";
+let urlParameters = new URLSearchParams(window.location.search);  
+
+var platform = new EducationPlatform();
+platform.initialize(urlParameters, TOKEN_HANDLER_URL);
+
+// Some functions and variables are accessed  
+// by onclick - or similer - events
+// We need to use window.x = x for this to work
+window.fit = platform.fit.bind(platform);
+window.updateGutterVisibility = platform.updateGutterVisibility.bind(platform);
+window.runAction = platform.runAction.bind(platform);
+window.panels = platform.panels;
+window.savePanelContents = platform.savePanelContents.bind(platform);
+window.toggle = platform.toggle.bind(platform);
+window.togglePanelById = platform.togglePanelById.bind(platform);
+//window.renderDiagram = renderDiagram;
+window.longNotification = platform.longNotification.bind(platform);
+window.getPanelTitle = platform.getPanelTitle.bind(platform);
```

Comment:
I guess it would be cleaner to rewrite those onClick handlers to access the platform object and interact with it directly...

Reply:
I think it would be better to not use window globals by registering these in the platform class but this should be a separate change.

I would prefer if the platform object were not added to the windows globals as there is the potential for it to clash and be modified inadvertently by other libraries. It would be good to namepace these current ones to something like `mdenetEducationPlatform`

Reply:
I see. Would be good to add an issue to clean this up: really there should be a way for action definitions to specify the languages to be used in output panels and that should be respected -- these output languages and their highlighting rules should then perhaps be loaded from a tool definition, though some might be provided centrally via the common tools specification.

Reply:
That sounds like an even better solution. I agree that we shouldn't fix this here, but we should have an issue that allows us to track the problem and prioritise it separately.

Reply:
Raised issue https://github.com/mdenet/educationplatform/issues/196 to capture this.